### ISSUE TITLE
Potential error in README

### ISSUE THREAD

**Issue by datamancy** [smbache/magrittr#98]
Hi all,

I think the example in the README under feature > basic piping
```
x %>% f %>% g %>% h = h(g(f(x))) 
```
is incorrect, here's an example where the above doesn't hold:
```
x %>% print %>% capture.output
```
This would only return the print value of x if it were true composition, instead it does partial evaluation, printing x and returning the printed value of x.

I'm not sure if this problem is worth solving, the additional writing to clarify the distinction may distract from the point.

**Comment by smbache** [comment]
This is really not an issue/error; by design magrittr will evaluate LHS before passing it to the next RHS, so non-standard evaluation, as with `capture.output` won't work as you try in the above. This design has several advantages in terms of implementation and also eg linearising otherwise "non-linear expressions":

```R
a %>% f %>% g(h(.)) # last RHS uses LHS twice, but evaluates it only once. 
```

**Comment by datamancy** [comment]
Hi Stefan,

Thanks for the quick reply, my point wasn't that this behaviour is wrong, it's that the README is slightly misleading. Before I ran into that behaviour the first time I had assumed, as the README states, that the pipe just composes the functions. Perhaps a footnote stating that it's not exactly equivalent to h(g(f(x))) and how would clarify the behaviour. Although if that would just muddy the point, it's probably not worth adding

**Comment by smbache** [comment]
Ahh ok, I see now. :) I'll reopen and keep that in mind to update. Thanks!

**Comment by smbache** [comment]
fixed in 00a1fe3305a4914d7c9714fba78fd5f03f70f51e
